modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogTableDescriptor.java

```diff
@@ -143,35 +107,35 @@ public CatalogTableDescriptor(
         this.colocationColumns = Objects.requireNonNullElse(colocationCols, pkCols);
         this.schemaVersions =  Objects.requireNonNull(schemaVersions, "No catalog schema versions.");
         this.storageProfile = Objects.requireNonNull(storageProfile, "No storage profile.");
+
+        TableVersion latestTableVersion = Objects.requireNonNull(schemaVersions.get(schemaVersions.latestVersion()));
+        if (!Objects.equals(latestTableVersion.columns(), columns)) {
+            throw new IllegalArgumentException("Latest schema version columns do not match descriptor definition columns.");
```

Comment:
Add the context to the exception message like: ".....,  latestTableVersion.columns = [...], schema columns=[...]"